src/editor.rs

```diff
@@ -7,335 +7,337 @@ use std::collections::hash_map::DefaultHasher;
 use std::hash::Hasher;
 use tuikit::prelude::*;
 
-mod command_log {
-    //! A utility datastructure to store and render a log of commands.  This is mostly used to give
-    //! the viewers of my streams feedback for what I'm typing.
-
-    use super::ActionCategory;
-    use tuikit::prelude::*;
-
-    /// Returns the [`Color`] that all [`Action`]s of a given [`ActionCategory`] should be
-    /// displayed.  This is not implemented as a method on [`ActionCategory`], because doing so
-    /// would require [`ActionCategory`] to rely on the specific terminal backend used.  This way,
-    /// we keep the terminal backend as encapsulated as possible.
-    pub fn term_color(category: ActionCategory) -> Color {
-        match category {
-            ActionCategory::Move => Color::LIGHT_BLUE,
-            ActionCategory::History => Color::LIGHT_YELLOW,
-            ActionCategory::Insert => Color::LIGHT_GREEN,
-            ActionCategory::Replace => Color::CYAN,
-            ActionCategory::Delete => Color::RED,
-            ActionCategory::Quit => Color::MAGENTA,
-            ActionCategory::Undefined => Color::LIGHT_RED,
-        }
-    }
-
-    /// One entry in the log.  This usually represts a single command, but could represent an
-    /// accumulation of many identical commands that are executed consecutively.
-    struct Entry {
-        count: usize,
-        command: String,
-        description: String,
-        color: Color,
-    }
-
-    /// A utility struct to store and display a log of which commands have been executed recently.
-    /// This is mostly used to give the viewers of my streams feedback for what I'm typing.
-    pub struct CommandLog {
-        /// A list of commands that have been run
-        commands: Vec<Entry>,
-        max_entries: usize,
-    }
-
-    impl CommandLog {
-        /// Create a new (empty) command log
-        pub fn new(max_entries: usize) -> CommandLog {
-            CommandLog {
-                commands: vec![],
-                max_entries,
+pub mod normal_mode{
+    pub mod command_log {
```

Comment:
You know, I wonder if it'd be worth moving `normal_mode` into it's own file - I think it'd be cleaner on the basis that `editor` will over time get more and more functionality.

This would entail moving the contents of `pub mod normal_mode` into `src/editor/normal_mode.rs`, and then moving the remainder of `src/editor.rs` to `src/editor/mod.rs` (Rust will refer to the contents of `src/folder_name/mod.rs` as though it were `src/folder_name.rs`).  We'd still need to put `pub mod normal_mode;` at the top of `src/editor/mod.rs`, though.  Would you be up for that?

Reply:
Sure.